src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlBulkCopy.cs

```diff
@@ -198,6 +194,10 @@ public SourceColumnMetadata(ValueMethod method, bool isSqlType, bool isDataFeed)
         private DataRowState _rowStateToSkip;
         private IEnumerator _rowEnumerator;
 
+        #if NETFRAMEWORK
+        private bool _rowSourceIsSqlDataReaderSmi;
+        #endif
```

Comment:
SqlDataReaderSmi can only be instantiated from context connections, so this will always be false on netfx. That's got knock-on implications throughout the file.

Reply:
You are right, and it's being tackled as part of part 2 of SMI dead code deletion 🚮